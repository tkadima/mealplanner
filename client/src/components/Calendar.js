import { Table } from 'semantic-ui-react'
import Meal from './Meal'

const Calendar = ({data, daysOfWeek, meals, handleSelectMeal, summary}) => {
    const headerRow = daysOfWeek.map((day, i) => 
        <Table.HeaderCell key={i}>{day}</Table.HeaderCell>
    )
    return (
            <Table celled className="Calendar">
                <Table.Header>
                    <Table.Row>
                        <Table.HeaderCell></Table.HeaderCell>
                        {headerRow}
                    </Table.Row>
                </Table.Header>
                <Table.Body>
                {
                    meals.map((meal, i) => {
                        const row = <Table.Row key={i}>
                            <Table.HeaderCell>{meal}</Table.HeaderCell>
                            {daysOfWeek.map((day, j) => {
                                let recipesForDayMeal = data[day][meal]
                                return (
                                    <Meal key={i + j} 
                                          meal={meal} 
                                          day={day} 
                                          selectMeal={handleSelectMeal}
                                          recipes={recipesForDayMeal}/>)
                                    })}
                        </Table.Row>
                        return row;
                    })
                }
                </Table.Body>
            </Table>
            
           
    )
}
export default Calendar
