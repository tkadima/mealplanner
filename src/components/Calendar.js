import { Table } from 'semantic-ui-react'
import Meal from './Meal'

const Calendar = ({data, daysOfWeek, meals, handleSelectMeal, summary}) => {
    const headerRow = daysOfWeek.map((day, i) => 
        <Table.HeaderCell key={i}>{day}</Table.HeaderCell>
    )
    const getTotal = (groupName, daySummary) => {
        return daySummary && daySummary[groupName] ? daySummary[groupName] : 0
    }
    const totalRow = daysOfWeek.map((day, i) => {
    //    let daySummary = summary[day]
    //     return <Table.Cell key={i}>
    //         <p color="blue">{getTotal('Dairy', daySummary)}/2 cups of dairy</p>
    //         <p color="red">{getTotal('Fruit', daySummary)}/1.5 cups of fruit</p>
    //         <p color="orange">{getTotal('Grain', daySummary)}/6 ounces of grain</p>
    //         <p color="purple">{getTotal('Protein', daySummary)}/5 ounces of protein</p>
    //         <p color="green">{getTotal('Vegetable', daySummary)}/2.5 cups of vegetables</p>
    //     </Table.Cell>
    })
   
    return (
            <Table celled className="Calendar">
                <Table.Header>
                    <Table.Row>
                        <Table.HeaderCell></Table.HeaderCell>
                        {headerRow}
                    </Table.Row>
                </Table.Header>
                <Table.Body>
                {
                    meals.map((meal, i) => {
                        const row = <Table.Row key={i}>
                            <Table.HeaderCell>{meal}</Table.HeaderCell>
                            {daysOfWeek.map((day, j) => {
                                let recipesForDayMeal = data[day][meal]
                                return (
                                    <Meal key={i + j} 
                                          meal={meal} 
                                          day={day} 
                                          selectMeal={handleSelectMeal}
                                          recipes={recipesForDayMeal}/>)
                                    })}
                        </Table.Row>
                        return row;
                    })
                }
                </Table.Body>
                <Table.Row>
                    <Table.Cell>Totals</Table.Cell>
                    {totalRow}
                </Table.Row>
            </Table>
            
           
    )
}
export default Calendar
