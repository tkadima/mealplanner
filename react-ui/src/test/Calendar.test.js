import Calendar from '../components/Calendar'
import { shallow } from 'enzyme'

test('should test Calendar component', () => {
    const daysOfWeek = ['Sunday']
    const meals = ['Breakfast', 'Lunch', 'Dinner']

    var initialProps = {
        data: {'Sunday': {'Breakfast':[], 'Lunch':[], 'Dinner':[] }},
        daysOfWeek: daysOfWeek, 
        meals:meals,
        handleSelectMeal:jest.fn() 
    }
    const wrapper = shallow(<Calendar {...initialProps}/>);
    expect(wrapper).toMatchSnapshot();
   });