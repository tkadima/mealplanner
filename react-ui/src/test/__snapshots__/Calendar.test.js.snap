// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test Calendar component 1`] = `ShallowWrapper {}`;
